tensorflow/core/grappler/optimizers/generic_layout_optimizer.cc

```diff
@@ -109,11 +109,19 @@ inline std::pair<string, string> GetSrcAndDstDataFormats(
     const TransposeContext& context, int num_gpus, int num_voltas) {
   string src_format = kNHWC;
   string dst_format = kNCHW;
-  if (((static_cast<float>(num_voltas) / static_cast<float>(num_gpus)) >=
-       kVoltaGPURatioThreshold) &&
-      NumConvOnDeviceWithDataTypeOverThreshold(context, kGPU, DT_HALF)) {
+
+  const bool is_NHWC_enforced = (!context.enforced_layout.empty()
+                                  && context.enforced_layout == "NHWC");
+  const bool should_swap =
+          ((static_cast<float>(num_voltas) / static_cast<float>(num_gpus))
+            >= kVoltaGPURatioThreshold)
+          && NumConvOnDeviceWithDataTypeOverThreshold(context, kGPU, DT_HALF);
+  // We swap only if NHWC is enforced or no layout is enforced
+  // and the devices config meet the thresholds
```

Comment:
Can you try to move words to the previous comment line until it hits the 80 char limit?
Please also squash your commits into one.

Reply:
Done